src/Hazelcast.Net/Core/TaskParallelExtensions.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Hazelcast.Core
+{
+    internal static class TaskParallelExtensions
```

Comment:
See my comment in the places where this is used. `Parallel.ForEachAsync` has been introduced in .NET 6 and we should just backport / polyfill it but use the same syntax, names and concepts. So... that class probably goes away.